jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -68,11 +68,18 @@ def parse_data(app, file_obj, data_type=None, data_label=None,
     #  generic enough to work with other file types (e.g. ASDF). For now, this
     #  supports MaNGA and JWST data.
     if isinstance(file_obj, fits.hdu.hdulist.HDUList):
-        _parse_hdulist(
-            app, file_obj, file_name=data_label,
-            flux_viewer_reference_name=flux_viewer_reference_name,
-            uncert_viewer_reference_name=uncert_viewer_reference_name
-        )
+        try:
+            _parse_spectrum1d_3d(
+                app, Spectrum1D.read(file_obj), data_label=data_label,
+                flux_viewer_reference_name=flux_viewer_reference_name,
+                uncert_viewer_reference_name=uncert_viewer_reference_name
+            )
+        except Exception:  # nosec
```

Comment:
can we do anything more specific than general exception that won't result in catching generic typos, etc?

Reply:
@rosteen , any idea? I don't think there is a way to know if any arbitrary input is going to succeed/fail the specutils I/O registry until you actually do it.

Reply:
but is there even a specific extension we can catch instead of the catch-all?

Reply:
If you mean exception, I purposely make it general here to be backward compatible. If specutils cannot parse *for whatever reason*, it goes back to the old way (until we can get rid of it because I think we should).

Reply:
"But is there a cleaner solution?"

Yes, there is. We can write a custom parser for each format that specutils cannot read. But I don't have that list or time.

Reply:
My concern is just that some random bug or typo could result in the except always being triggered and we have no way of knowing that or noticing it in the tests.  Maybe for now just a snackbar if the except is entered and then could check for that snackbar message in test coverage could do?

Reply:
I do not see how that is worse than what we already have. Currently, it goes through our own HDU parser anyway. With this change, it would still do that and then still crash like it would currently if both specutils and HDU parser cannot handle it. The only difference introduced by this PR is that it tries specutils first. A snackbar message before fallback would just be confusing if it ends up succeeding later.

Reply:
agreed from the user-perspective, I'm more concerned about testing and ending up with stale code without noticing (or just not clear which path a file took).  But as long as there is a plan to consolidate these later, I guess this is fine for now.  Thanks!

Reply:
I think @rosteen is going to end up refactoring this for his specutils 2.0 work?